docs/DeveloperGuide.adoc

```diff
@@ -176,94 +176,214 @@ through all the stored data and generate the statistics data whenever it is quer
+by the event.
 
-Given below is an example usage scenario and how the undo/redo mechanism behaves at each step.
+Given below is an example usage scenario and how the auto allocation mechanism behaves at each step.
 
-Step 1. The user launches the application for the first time. The `VersionedAddressBook` will be initialized with the initial address book state, and the `currentStatePointer` pointing to that single address book state.
+._Program flow of the Auto Allocate Feature_
+image::AutoAllocateFlowChart.png[]
 
-image::UndoRedoState0.png[]
+**Step 1**. The user executes `allocate 1 n/2 t/female` with the intention to allocate 2 employees with tag [female]
+to the 1st event displayed in the event list.
+
+**Step 2**. The command checks if `eventIndex` is valid and if `ManpowerCountToAdd` is specified.
+[NOTE]
+If `ManpowerCountToAdd` is not specified, it is assumed to be the maximum number possible for the event.
+Validity of other command arguments e.g. if `ManpowerCountToAdd` is a positive integer is checked
+by `AutoAllocateCommandParser` and not within the command `AutoAllocateCommand`.
 
-Step 2. The user executes `delete 5` command to delete the 5th employee in the address book. The `delete` command calls `Model#commitAddressBook()`, causing the modified state of the address book after the `delete 5` command executes to be saved in the `addressBookStateList`, and the `currentStatePointer` is shifted to the newly inserted address book state.
 
-image::UndoRedoState1.png[]
+**Step 3**. The command calls its own method `AutoAllocateCommand#getManpowerNeededByEvent()` to get the number of employees
+required by the specified event.
 
-Step 3. The user executes `add n/David ...` to add a new employee. The `add` command also calls `Model#commitAddressBook()`, causing another modified address book state to be saved into the `addressBookStateList`.
 
-image::UndoRedoState2.png[]
+**Step 4**. The command calls its own method `AutoAllocateCommand#createAvailableEmployeeListForEvent()` to create a
+filtered list of employees based on the `tagList` and if employee satisfies `Event#isAvailableForEvent()`.
 
+**Step 5**. The command checks if supply (number of employees in filtered list in step 4) exceeds
+demand (number of employees required by event, generated in step 3).
 [NOTE]
-If a command fails its execution, it will not call `Model#commitAddressBook()`, so the address book state will not be saved into the `addressBookStateList`.
+If supply < demand, throws exception to user. If supply > demand, employees will be randomly selected instead.
+
+**Step 6**. The command calls `Event#allocateEmployee()` to allocate the chosen employees to the event.
+
+[NOTE]
+For storage purposes, only the `Employee#EmployeeId` is allocated to the event.
+
+**Step 7**. Done.
+
+
+The following sequence diagram shows how the auto allocation works:
+
+._Sequence Diagram of the AutoAllocate Command_
```

Comment:
Your return arrow in the all the self-invocation calls should be dotted. The return arrow from AutoAllocateCommandParser and AddressBookParser should be dotted too.

<img width="117" alt="Screen Shot 2019-10-24 at 1 39 20 PM" src="https://user-images.githubusercontent.com/44959941/67456565-c8a59200-f663-11e9-9a0d-a6d8f4bcb021.png">

Reply:
thanks thanks